pkg/plugins/plugins.go

```diff
@@ -0,0 +1,24 @@
+// Copyright 2020 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+package plugins
+
+import (
+	"github.com/newrelic/infrastructure-agent/internal/agent"
+	"github.com/newrelic/infrastructure-agent/pkg/config"
+	"github.com/newrelic/infrastructure-agent/pkg/plugins/ids"
+)
+
+func isHeartbeatOnlyMode(cfg *config.Config) bool {
+	return cfg.IsSecureForwardOnly || cfg.IsIntegrationsOnly
+}
+
+func registerIntegrationsOnlyPlugin(agt *agent.Agent) {
+	cfg := agt.GetContext().Config()
+	if cfg.IsSecureForwardOnly {
```

Comment:
just a nit, I don't know if there is a config check at initialization but these seems to be a good candidate for doing it when first parsing the config and probably crashing , so is clear that this config is not supported

Reply:
Sure, I will check it

Reply:
I am not sure if I would move it there, there is certainly a NormalizeConfig function on the config.go where we could be placing that if both are set, to unset one of them, but I feel it's good to have the control here, since you can create a config without calling the NormalizeConfig function.